feat: add minimal progress logging to pipeline components

yahgw-ai opened the pull request:
## Summary
- Added minimal console.log statements to provide visibility into pipeline execution progress
- Helps users track which component is running and understand if the process is stuck
- No verbose output that would clutter logs or impact performance

## Changes
- **Orchestrator**: Logs pipeline start, component transitions, and completion
- **BlockFinder**: Logs date range being processed and each date as it's processed
- **DistributorDetector**: Logs scan range with dates, new distributors found, and chunk progress
- **BalanceFetcher**: Logs distributor progress (X/Y) and each date fetch
- **RecipientRecievedScanner**: Logs distributor progress, date range per distributor, and event count

## Test Plan
- [x] Run the pipeline and verify progress logs appear at expected points
- [x] Verify log messages are concise and informative
- [x] Ensure no performance impact from logging
- [ ] Fix the failing test in distributor-detector that expects readBlockNumbers to be called once (now called twice due to logging)

Resolves #247

yahgw-ai committed:
feat: add minimal progress logging to pipeline components

Add console.log statements to provide visibility into pipeline execution:

- Orchestrator: log pipeline start, component transitions, and completion
- BlockFinder: log date range and individual date processing
- DistributorDetector: log scan range, new distributors, and chunk progress
- BalanceFetcher: log distributor and date progress
- RecipientRecievedScanner: log distributor progress, date range, and event counts

Resolves #247

yahgwai commented on src/distributor-detector.ts:
```diff
@@ -221,6 +223,27 @@ export class DistributorDetector {
     const endBlock = this.getBlockForDate(endDate);
     const scanRange = this.calculateScanRange(existingData, endBlock);
 
+    // Get date for fromBlock for logging
```
Okay, don't bother with this section. It's not worth the additional logic to find the date.

yahgw-ai committed:
fix: simplify distributor-detector logging per PR feedback

Remove complex date lookup logic from distributor-detector logging as requested by @yahgwai.
The logging now simply shows the block range without trying to determine dates.
